tests/schema/test_enum.py

```diff
@@ -100,6 +110,26 @@ def eat_cone(self, input: ConeInput) -> bool:
     assert result.data["eatCone"] is True
 
 
+def test_lazy_enum_arguments():
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def something(self, enum: LazySharedEnum) -> LazySharedEnum:
```

Comment:
can you add a test for `Annotated[SharedEnum, strawberry.lazy()`?

we deprecated LazyType in favour of the new syntax https://strawberry.rocks/docs/types/lazy#lazy-types

Reply:
Actually that triggers a new bug :disappointed: 

This seems unrelated to the original bug and I'll leave it to a separate PR.

In summary:

When using `LazySharedEnum = Annotated["SharedEnum", lazy(__name__)]`:
- The argument type in the function signature is `Annotated[ForwardRef("SharedEnum"), lazy(__name__)]`
- `ReservedType.find()` calls `_eval_type()` and converts it into `Annotated[SharedEnum, lazy(__name__)]`
- `StrawberryArgument._parse_annotated()` calls `StrawberryLazyReference.resolve_forward_ref( SharedEnum )`. It crashes because it is not a ForwardRef

Also, I'm not sure what the proper fix would be:

- Should we ignore the current module and load it from the `lazy()` hint?
- Should it ignore the lazy() hint and load from the current module()?
- Should it be an error to apply `lazy()` annotation if the class can be resolved on the current module?
- Should it be a no-op to define `Annotated["ClassName", lazy("current_module")]` ?

Reply:
It works fine if the enum is actually defined in a different file, I've updated the test to use it